services/w3c/w3c-validation-helper.spec.js

```diff
@@ -0,0 +1,381 @@
+      {
+        preset: 'HTML, SVG 1.1, MathML 3.0, RDFa Lite 1.1',
+        expectedResult: true,
+      },
+      {
+        preset: 'HTML , SVG 1.1 , MathML 3.0 , RDFa Lite 1.1',
+        expectedResult: true,
+      },
+      {
+        preset: 'HTML 4.01 Strict,URL/XHTML 1.0 Strict,URL',
+        expectedResult: true,
+      },
+      {
+        preset: 'HTML 4.01 Strict, URL/ XHTML 1.0 Strict, URL',
+        expectedResult: true,
+      },
+      {
+        preset: 'HTML 4.01 Strict , URL / XHTML 1.0 Strict , URL',
+        expectedResult: true,
+      },
+      {
+        preset: 'HTML 4.01 Transitional,URL/XHTML 1.0 Transitional,URL',
+        expectedResult: true,
+      },
+      {
+        preset: 'HTML 4.01 Transitional, URL/ XHTML 1.0 Transitional, URL',
+        expectedResult: true,
+      },
+      {
+        preset: 'HTML 4.01 Transitional , URL / XHTML 1.0 Transitional , URL',
+        expectedResult: true,
+      },
+      {
+        preset: 'HTML 4.01 Frameset,URL/XHTML 1.0 Frameset,URL',
+        expectedResult: true,
+      },
+      {
+        preset: 'HTML 4.01 Frameset, URL/ XHTML 1.0 Frameset, URL',
+        expectedResult: true,
+      },
+      {
+        preset: 'HTML 4.01 Frameset , URL / XHTML 1.0 Frameset , URL',
+        expectedResult: true,
+      },
+      { preset: 'XHTML,SVG 1.1,MathML 3.0', expectedResult: true },
+      { preset: 'XHTML, SVG 1.1, MathML 3.0', expectedResult: true },
+      { preset: 'XHTML , SVG 1.1 , MathML 3.0', expectedResult: true },
+      {
+        preset: 'XHTML,SVG 1.1,MathML 3.0,RDFa Lite 1.1',
+        expectedResult: true,
+      },
+      {
+        preset: 'XHTML, SVG 1.1, MathML 3.0, RDFa Lite 1.1',
+        expectedResult: true,
+      },
+      {
+        preset: 'XHTML , SVG 1.1 , MathML 3.0 , RDFa Lite 1.1',
+        expectedResult: true,
+      },
+      {
+        preset: 'XHTML 1.0 Strict,URL,Ruby,SVG 1.1,MathML 3.0',
+        expectedResult: true,
+      },
+      {
+        preset: 'XHTML 1.0 Strict, URL, Ruby, SVG 1.1, MathML 3.0',
+        expectedResult: true,
+      },
+      {
+        preset: 'XHTML 1.0 Strict , URL , Ruby , SVG 1.1 , MathML 3.0',
+        expectedResult: true,
+      },
+      { preset: 'SVG 1.1,URL,XHTML,MathML 3.0', expectedResult: true },
+      { preset: 'SVG 1.1, URL, XHTML, MathML 3.0', expectedResult: true },
+      { preset: 'SVG 1.1 , URL , XHTML , MathML 3.0', expectedResult: true },
+      { preset: undefined, expectedResult: false },
+      { preset: null, expectedResult: false },
+      { preset: '', expectedResult: false },
+      { preset: 'html', expectedResult: false },
+    ]
+
+    presetTests.forEach(test => {
+      // Arrange
+      const { preset, expectedResult } = test
+      it(`should return ${expectedResult} for ${preset}`, function() {
+        // Act
+        const actualResult = presetRegex.test(preset)
+
+        // Assert
+        expect(actualResult).to.equal(expectedResult)
+      })
+    })
+  })
```

Comment:
Cheers for following the Arrange/Act/Assert pattern, though we should remove the explicit comments/spacing calling it out in these tests.

Reply:
Sure I will do that now